internal/cmd/stack/environment.go

```diff
@@ -0,0 +1,157 @@
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/shurcooL/graphql"
+	"github.com/urfave/cli/v2"
+
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+)
+
+// ConfigType is a type of configuration element.
+type ConfigType string
+
+// ConfigInput represents the input required to create or update a config
+// element.
+type ConfigInput struct {
+	ID        graphql.ID      `json:"id"`
+	Type      ConfigType      `json:"type"`
+	Value     graphql.String  `json:"value"`
+	WriteOnly graphql.Boolean `json:"writeOnly"`
+}
+
+func setVar(cliCtx *cli.Context) error {
+	if nArgs := cliCtx.NArg(); nArgs != 2 {
+		return fmt.Errorf("expecting environment setenv as two arguments, got %d instead", nArgs)
+	}
+
+	envName := cliCtx.Args().Get(0)
+	envValue := cliCtx.Args().Get(1)
+
+	stackID := cliCtx.String(flagStackID.Name)
+
+	var mutation struct {
+		ConfigElement struct {
+			ID        string `graphql:"id"`
+			WriteOnly bool   `graphql:"writeOnly"`
+			Value     string `graphql:"value"`
+		} `graphql:"stackConfigAdd(stack: $stack, config: $config)"`
+	}
+
+	variables := map[string]interface{}{
+		"stack": graphql.ID(stackID),
+		"config": ConfigInput{
+			ID:        graphql.ID(envName),
+			Type:      ConfigType("ENVIRONMENT_VARIABLE"),
+			Value:     graphql.String(envValue),
+			WriteOnly: graphql.Boolean(cliCtx.Bool(flagEnvironmentWriteOnly.Name)),
+		},
+	}
+
+	if err := authenticated.Client.Mutate(context.Background(), &mutation, variables); err != nil {
+		return err
+	}
+
+	fmt.Printf("Environment variable (%s) has been set!\n", mutation.ConfigElement.ID)
+	if mutation.ConfigElement.WriteOnly {
+		fmt.Printf("Value: %s \n", strings.Repeat("*", len(envValue)))
+	} else {
+		fmt.Printf("Value: %s \n", mutation.ConfigElement.Value)
+	}
+	fmt.Printf("Write only: %t \n", mutation.ConfigElement.WriteOnly)
+
+	return nil
+}
+
+func mountFile(cliCtx *cli.Context) error {
+	nArgs := cliCtx.NArg()
+
+	envName := cliCtx.Args().Get(0)
+	stackID := cliCtx.String(flagStackID.Name)
+
+	var fileContent []byte
+	var err error
+
+	switch nArgs {
+	case 1:
+		fmt.Println("Reading from STDIN...")
+
+		if fileContent, err = ioutil.ReadAll(os.Stdin); err != nil {
+			return fmt.Errorf("couldn't read from STDIN: %w", err)
+		}
+	case 2:
+		filePath := cliCtx.Args().Get(1)
+
+		if fileContent, err = os.ReadFile(filePath); err != nil {
+			return fmt.Errorf("couldn't read file from %s: %w", filePath, err)
+		}
+	default:
+		return fmt.Errorf("expecting environment mount as max two arguments, got %d instead", nArgs)
+	}
+
+	var mutation struct {
+		ConfigElement struct {
+			ID        string     `graphql:"id"`
+			WriteOnly bool       `graphql:"writeOnly"`
+			Type      ConfigType `graphql:"type"`
+		} `graphql:"stackConfigAdd(stack: $stack, config: $config)"`
+	}
+
+	variables := map[string]interface{}{
+		"stack": graphql.ID(stackID),
+		"config": ConfigInput{
+			ID:        graphql.ID(envName),
```

Comment:
Should we do any kind of validation to make sure this is a valid unix path?

Reply:
We do this on the server side.